### PR TITLE
Update News Feed and Preferences widget

### PR DESCRIPTION
**Pull request by kmshin1397**
I noticed that the newsfeed GET API was returning duplicate items for sources added because each source gets a row for every group it is a part of in the Sources table. This PR makes it so the API filters for distinct source additions by time.

### CODE REVIEW

**Comment by kmshin1397** [comment]
Screenshot 2020-09-03 103727

File: skyportal/tests/frontend/test_newsfeed.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import uuid
 import pytest
-
+from selenium.webdriver.common.action_chains import ActionChains
```
Comment (dmitryduev) [inline skyportal/tests/frontend/test_newsfeed.py:3]: Neat, going to start using this!

**Review by dmitryduev — APPROVED** [review]
LGTM! `test_news_feed_prefs_widget` is flaking - have you looked into it?

File: skyportal/tests/frontend/test_newsfeed.py (hunk 1 above)
Comment (kmshin1397) [reply skyportal/tests/frontend/test_newsfeed.py:3]: Yeah starting to really look into it now... Using ActionChains fixed some flakiness in test_groups for me yesterday but doesn't seem like it's helping here...

**Comment by kmshin1397** [comment]
Okay, this was originally just a quick tweak to the backend news feed query, but in the process of trying to get test_news_feed_prefs_widget green I ended up doing some reskinning of the preferences widget as well.

Basically, I couldn't get it green so I tried just refactoring the preferences to use MUI components and that at least got Travis to green (without using .pause() either).

New preferences widget look:
Screenshot 2020-09-03 155421

@dmitryduev @stefanv

**Comment by stefanv** [comment]
Nice work.  An enemy of pause is a friend of mine.

**Pull request merged**